client/signup/steps/website-content/page-details.tsx

```diff
@@ -36,7 +35,7 @@ export function PageDetails( {
 } ) {
 	const translate = useTranslate();
 	const dispatch = useDispatch();
-	const site = useSelector( getSelectedSite );
+	const site = useSelector( getSelectedSite ) ?? undefined;
```

Comment:
This prevents `site` from being `null` and makes it `undefined` instead, which is supported by `WordpressMediaUpload`.